drivers/media/platform/bcm2835/bcm2835-unicam.c

```diff
@@ -2369,13 +2420,21 @@ static int register_node(struct unicam_device *unicam, struct unicam_node *node,
 
 static void unregister_nodes(struct unicam_device *unicam)
 {
-	if (unicam->node[IMAGE_PAD].registered) {
-		video_unregister_device(&unicam->node[IMAGE_PAD].video_dev);
-		unicam->node[IMAGE_PAD].registered = 0;
-	}
-	if (unicam->node[METADATA_PAD].registered) {
-		video_unregister_device(&unicam->node[METADATA_PAD].video_dev);
-		unicam->node[METADATA_PAD].registered = 0;
+	struct unicam_node *node;
+	int i;
+
+	for (i = 0; i < MAX_NODES; i++) {
+		node = &unicam->node[i];
+		if (node->registered) {
```

Comment:
Done the video_register_device failure case.

Done the list ctrl on node 1 fix.  The ctrl_handler was set globally for the driver and not per node.  I had actually changed that some time back, but that change seems to have gotten lost in the process.

Regarding the bind/unbind, not done anything for this yet, but I wonder if the ctrl_handler registration should be moved into node open() instead of probe?

Reply:
> Regarding the bind/unbind, not done anything for this yet, but I wonder if the ctrl_handler registration should be moved into node open() instead of probe?

Probably, but I'm not going to worry for now. I've had enough of reading through this driver! I'd need to investigate how unbind is triggered anyway. IIRC Unicam will take a reference on the subdev (or the other way around), so you have to unload them in the correct order, which I think is Unicam first.